### PR TITLE
[Tests] [JSight to OpenAPI converter] req.japi.type_0.3

### PR DESCRIPTION
**Pull request by bereg2k**
- Added new tests **for the requirement**: `req.japi.type_0.3`
- Added new tests **into the existing test files**: `regex_notation_in_directive_type`

### CODE REVIEW

File: testdata/jsight_0.3/ser-to-openapi/req.japi.type_0.3/user_type_or.openapi.todo [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            "type": "number",
+            "example": 3.0004
+          }
+        },
+        "required": [
+          "three"
+        ],
+        "additionalProperties": false
+      },
+      "userType1SelfRef": {
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.japi.type_0.3/user_type_or.openapi.todo:48]: Should be `userType1Self`

**Review by andrew-platonov — CHANGES_REQUESTED** [review]

File: testdata/jsight_0.3/ser-to-openapi/req.japi.type_0.3/user_type_basic.openapi.todo [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "openapi": "3.0.3",
+  "info": {
+    "title": "",
+    "version": ""
+  },
+  "components": {
+    "schemas": {
+      "cat": {
```
[the remark this answers is not in the document]
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.type_0.3/user_type_basic.openapi.todo:9]: Replaced with `userTypeBasic`

File: testdata/jsight_0.3/ser-to-openapi/req.japi.schema_notation/req.japi.schema_notation.regex_0.3/regex_notation_in_directive_type.openapi.todo [hunk 3]
```diff
@@ -68,6 +80,26 @@
                 }
               }
             }
+          },
+          "500": {
+            "description": "",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "oneOf": [
```
[the remark this answers is not in the document]
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.schema_notation/req.japi.schema_notation.regex_0.3/regex_notation_in_directive_type.openapi.todo:89]: Understood, thanks! 

Fixed here.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.type_0.3/user_type_or.openapi.todo (hunk 1 above)
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.type_0.3/user_type_or.openapi.todo:48]: Fixed.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.type_0.3/user_type_basic.openapi.todo (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrew-platonov) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.type_0.3/user_type_basic.openapi.todo]: The content of the object is also wrong)

[the remark this answers is not in the document]
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.type_0.3/user_type_basic.openapi.todo]: Fixed.

**Pull request merged**